sbi/inference/posteriors/score_posterior.py

```diff
@@ -0,0 +1,393 @@
+                x_o=None,
+                enable_transform=enable_transform,
+            )
+        )
+
+        device = device if device is not None else potential_fn_gradient.device
+
+        self.prior = prior
+        self.score_estimator = score_estimator
+        self.potential_fn_gradient = potential_fn_gradient
+        self.theta_transform = theta_transform
+
+        self.max_sampling_batch_size = max_sampling_batch_size
+
+        self._purpose = """It samples from the diffusion model given the \
+            score_estimator."""
+
+    def sample(
+        self,
+        sample_shape: Shape = torch.Size(),
+        x: Optional[Tensor] = None,
+        predictor: Union[str, Predictor] = "euler_maruyama",
+        corrector: Optional[Union[str, Corrector]] = None,
+        predictor_params: Optional[Dict] = None,
+        corrector_params: Optional[Dict] = None,
+        steps: int = 500,
+        ts: Optional[Tensor] = None,
+        max_sampling_batch_size: int = 10_000,
+        sample_with: Optional[str] = None,
+        show_progress_bars: bool = True,
+    ) -> Tensor:
+        r"""Return samples from posterior distribution $p(\theta|x)$.
+
+        Args:
+            sample_shape: Shape of the samples to be drawn.
+            x: Deprecated - use `.set_default_x()` prior to `.sample()`.
+            method: Method to use for sampling. Currently, only "euler_maruyma" is
+                supported.
+            steps: Number of steps to take for the Euler-Maruyama method.
+            max_sampling_batch_size: Maximum batch size for sampling.
+            sample_with: Deprecated - use `.build_posterior(sample_with=...)` prior to
+                `.sample()`.
+            show_progress_bars: Whether to show a progress bar during sampling.
+        """
+
+        num_samples = torch.Size(sample_shape).numel()
+
+        x = self._x_else_default_x(x)
+        x = reshape_to_batch_event(x, self.score_estimator.condition_shape)
+        self.potential_fn_gradient.set_x(x)
+
+        max_sampling_batch_size = (
+            self.max_sampling_batch_size
+            if max_sampling_batch_size is None
+            else max_sampling_batch_size
+        )
+
+        if sample_with is not None:
+            raise ValueError(
+                f"You set `sample_with={sample_with}`. As of sbi v0.18.0, setting "
+                f"`sample_with` is no longer supported. You have to rerun "
+                f"`.build_posterior(sample_with={sample_with}).`"
+            )
+
+        if ts is None:
+            T_max = self.score_estimator.T_max
+            T_min = self.score_estimator.T_min
+            ts = torch.linspace(T_max, T_min, steps)
+
+        diffuser = Diffuser(
+            self.potential_fn_gradient, predictor=predictor, corrector=corrector
+        )
+        max_sampling_batch_size = min(max_sampling_batch_size, num_samples)
+        samples = []
+        num_iter = num_samples // max_sampling_batch_size
```

Comment:
rename to `num_batches`

Reply:
can  this be parallelized? or is the diffuser running sequentially over batches?

Reply:
I think this is the same as for NPE, we just iterate over very large batches.